plenoptic/metric/perceptual_distance.py

```diff
@@ -247,9 +246,97 @@ def ssim_map(img1, img2, dynamic_range=1):
        http://dx.doi.org/10.1167/8.12.8
 
     """
+    if min(img1.shape[2], img1.shape[3]) < 11:
+        warnings.warn("SSIM uses 11x11 convolutional kernel, but the height and/or "
+                      "the width of the input image is smaller than 11, so the "
+                      "kernel size is set to be the minimum of these two numbers.")
     return _ssim_parts(img1, img2, dynamic_range)[0]
 
 
+def ms_ssim(img1, img2, dynamic_range=1, power_factors=None):
+    r"""Multiscale structural similarity index (MS-SSIM)
+
+    As described in [1]_, multiscale structural similarity index (MS-SSIM) is
+    an improvement upon structural similarity index (SSIM) that takes into
+    account the perceptual distance between two images on different scales.
+
+    SSIM is based on three comparison measurements between the two images:
+    luminance, contrast, and structure. All of these are computed convolutionally
+    across the images, producing three maps instead of scalars. The SSIM map is
+    the elementwise product of these three maps. See `metric.ssim` and
+    `metric.ssim_map` for a full description of SSIM.
+
+    To get images of different scales, average pooling operations with kernel
+    size 2 are performed recursively on the input images. The product of
+    contrast map and structure map (the "contrast-structure map") is computed
+    for all but the coarsest scales, and the overall SSIM map is only computed
+    for the coarsest scale. Their mean values are raised to exponents and
+    multiplied to produce MS-SSIM:
+    .. math::
+        MSSSIM = {SSIM}_M^{a_M} \prod_{i=1}^{M-1} ({CS}_i)^{a_i}
+    Here :math: `M` is the number of scales, :math: `{CS}_i` is the mean value
+    of the contrast-structure map for the i'th finest scale, and :math: `{SSIM}_M`
+    is the mean value of the SSIM map for the coarsest scale. If at least one
+    of these terms are negative, the value of MS-SSIM is zero. The values of
+    :math: `a_i` are taken from the argument `power_factors`.
```

Comment:
does `a_M` also come from `power_factors`? And `M` and `i` are just two different indices?

Reply:
Oh `M` is the number of scales, which is written just under the formula. It's just an index like `i`. You can see that `i` goes from `1` to `M-1` in the product. Do you think this should be made clearer?

Reply:
oh okay, so `a_M` is the final value in `power_factors`, and `a_i` are the values up to that. I think I'd just say `The values of :math: a_i and a_M are taken from power_factors` or the like, but this isn't important.

Reply:
Ok I changed it to `a_i, i=1,...,M`

Reply:
works for me!